subcommands/targets/offline-update.go

```diff
@@ -0,0 +1,286 @@
+	subcommands.DieNotNil(err, "Failed to download Target's Apps:")
+
+	fmt.Println("Successfully downloaded offline update content")
+}
+
+func checkIfTargetExists(targetName string) (bool, error) {
+	data, err := api.TufMetadataGet(factory, "targets.json", tag, prod)
+	if herr := client.AsHttpError(err); herr != nil && herr.Response.StatusCode == 404 {
+		return false, nil
+	}
+	if err != nil {
+		return false, err
+	}
+
+	targets := client.AtsTufTargets{}
+	err = json.Unmarshal(*data, &targets)
+	if err != nil {
+		return false, err
+	}
+	for tn := range targets.Signed.Targets {
+		if tn == targetName {
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+func getTargetInfo(targetName string) (*targetInfo, error) {
+	// Getting Target's custom info from the `/ota/factories/<factory>/targets/<target-name>` because:
+	// 1. a target name is unique and represents the same Target across all "tagged" targets set including prod;
+	// 2. only this target version/representation contains an original tag(s)/branch that
+	// the `image-assemble` and apps fetching was performed for (needed for determining where to download Apps from).
+	custom, err := getTargetCustomInfo(targetName)
+	if err != nil {
+		return nil, err
+	}
+
+	info := targetInfo{}
+	info.version, err = strconv.Atoi(custom.Version)
+	if err != nil {
+		return nil, err
+	}
+	info.hardwareID = custom.HardwareIds[0]
+	info.buildTag = custom.Tags[0] // See the assemble.py script in ci-scripts https://github.com/foundriesio/ci-scripts/blob/18b4fb154c37b6ad1bc6e7b7903a540b7a758f5d/assemble.py#L300
+	info.ostreeVersion = info.version
+	if len(custom.OrigUri) > 0 {
+		indx := strings.LastIndexByte(custom.OrigUri, '/')
+		if indx == -1 {
+			fmt.Fprintf(os.Stderr, "Failed to determine Target's OE build version: %s", custom.OrigUri)
+			os.Exit(1)
+		}
+		targetOstreeVerStr := custom.OrigUri[indx+1:]
+		info.ostreeVersion, err = strconv.Atoi(targetOstreeVerStr)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return &info, nil
+}
+
+func downloadTufRepo(factory string, dstDir string) error {
+	ver := 1
+
+	downloadMetadataFile := func(metadataFileName string) error {
+		data, err := api.TufMetadataGet(factory, metadataFileName, tag, prod)
+		if err != nil {
+			return err
+		}
+		f, err := os.Create(path.Join(dstDir, metadataFileName))
+		if err != nil {
+			return err
+		}
+		defer f.Close()
+		_, err = f.Write(*data)
+		return err
+	}
+
+	err := os.MkdirAll(dstDir, 0755)
+	if err != nil {
+		return err
+	}
+
+	for {
+		metadataFileName := fmt.Sprintf("%d.root.json", ver)
+		err := downloadMetadataFile(metadataFileName)
+		if err != nil {
+			httpErr := client.AsHttpError(err)
+			if httpErr != nil {
+				// TODO: Fix an issue in ota-lite that returns 500 for non-existing non-prod N.root.json,
+				// and then remove 500 check here
+				if httpErr.Response.StatusCode != 404 && httpErr.Response.StatusCode != 500 {
+					return err
+				}
```

Comment:
Please, do us a favor - fix that ;-)
Thanks!

Reply:
Sure, but not in this PR :).

Reply:
I mean: please, fix that and update this PR.

Sorry for that bug, we just cannot ship with it as is :( .

Reply:
This bug is already there and is not a blocker for offline update at all.
It's not a big deal to fix it, but it takes time, so maybe we deliver this to the support/platform team at first so they start integration conversation with the customer. Meanwhile, I'll fix this issue (the fix in ota-lite is trivial, just testing doesn't work as I expected...)